### PR TITLE
Add some codeowners for `Tools/clinic/`

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -173,3 +173,8 @@ Doc/c-api/stable.rst          @encukou
 
 # zipfile.Path
 **/*zipfile/*_path.py         @jaraco
+
+# Argument Clinic
+/Tools/clinic/**              @erlend-aasland
+/Tools/clinic/mypy.ini        @AlexWaygood
+/Tools/clinic/requirements-dev.txt  @AlexWaygood
```
Comment (erlend-aasland) [inline .github/CODEOWNERS]: Did you consider this? 😃 

```suggestion
/Tools/clinic/**              @erlend-aasland @AlexWaygood
```

**Review by erlend-aasland — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (AlexWaygood) [reply .github/CODEOWNERS]: oh, go on then!

**Review by erlend-aasland — APPROVED** [review]
🚀

**Comment by AlexWaygood** [comment]
Worth backporting?

**Comment by erlend-aasland** [comment]
> Worth backporting?

I don't think it adds much value; PRs are normally made against `main` and then backported.

**Pull request merged**